### PR TITLE
feat(xmllib): expose lib through init and enable colored printing (RDU-39)

### PR DESCRIPTION
**Pull request by Nora-Olivia-Ammann**

### CODE REVIEW

File: src/dsp_tools/models/custom_warnings.py [hunk 1]
```diff
@@ -19,7 +24,16 @@ class DspToolsUserWarning(Warning):
     @classmethod
     def showwarning(cls, message: str) -> None:
         """Print the warning, without context"""
-        cprint(f"WARNING: {message}", color="red", attrs=["bold"])
+        print(BOLD_RED + f"WARNING: {message}" + RESET_TO_DEFAULT)
```
Comment (Nora-Olivia-Ammann) [inline src/dsp_tools/models/custom_warnings.py:29]: Interestingly it does not work with cprint. You have to configure them with ANSI, but I think this way it is tenable for future use.

File: src/dsp_tools/xmllib/__init__.py [hunk 2]
```diff
@@ -1,4 +1,6 @@
 from .helpers import create_label_to_name_list_node_mapping as create_label_to_name_list_node_mapping
+from .helpers import create_list_from_string as create_list_from_string
```
Comment (Nora-Olivia-Ammann) [inline src/dsp_tools/xmllib/__init__.py:2]: previously forgot to export them

File: src/dsp_tools/models/custom_warnings.py [hunk 3]
```diff
@@ -1,7 +1,20 @@
 from typing import Protocol
 from typing import runtime_checkable
 
-from termcolor import cprint
+# These are ANSI escape codes which can be used to configure the print output on the terminal
+# https://en.wikipedia.org/wiki/ANSI_escape_code
+# The semicolon separates different configurations
+
+# All codes must start with "\033["
```
Comment (Nora-Olivia-Ammann) [inline src/dsp_tools/models/custom_warnings.py:8]: I have taken it apart, so that for the future it will be easier to understand the construct.

Comment (jnussbaum) [reply src/dsp_tools/models/custom_warnings.py:8]: Very good introduction to ANSI codes, thanks :-)

File: src/dsp_tools/models/custom_warnings.py [hunk 4]
```diff
@@ -19,7 +32,16 @@ class DspToolsUserWarning(Warning):
     @classmethod
     def showwarning(cls, message: str) -> None:
         """Print the warning, without context"""
-        cprint(f"WARNING: {message}", color="red", attrs=["bold"])
+        print(BOLD_RED + f"WARNING: {message}" + RESET_TO_DEFAULT)
+
+
+class DspToolsUserInfo(Warning):
+    """Class for general user-facing warnings"""
```
Comment (jnussbaum) [inline src/dsp_tools/models/custom_warnings.py:39]: Identical docstring as above, perhaps you want to formulate one specific for this class?

File: src/dsp_tools/models/custom_warnings.py [hunk 5]
```diff
@@ -1,7 +1,20 @@
...
+# All codes must start with "\033["
+SEQUENCE_START = "\033["
+
+# the "m" at the end signals,
+# that the configuration code is finished and after that the string that should be printed starts
+SEQUENCE_END = "m"
+
+BOLD_RED = SEQUENCE_START + "1;31" + SEQUENCE_END  # 1 (bold) ; 31 (red)
+YELLOW = SEQUENCE_START + "0;33" + SEQUENCE_END  # 0 (normal font) ; 33 (yellow)
+RESET_TO_DEFAULT = SEQUENCE_START + "0" + SEQUENCE_END  # reset to the default setting of the console
```
Comment (jnussbaum) [inline src/dsp_tools/models/custom_warnings.py:17]: I would use f-strings consistently, and no `+` concatenation at all.

File: src/dsp_tools/models/custom_warnings.py [hunk 6]
```diff
@@ -19,7 +32,16 @@ class DspToolsUserWarning(Warning):
...
+        print(BOLD_RED + f"WARNING: {message}" + RESET_TO_DEFAULT)
+
+
+class DspToolsUserInfo(Warning):
+    """Class for general user-facing warnings"""
+
+    @classmethod
+    def showwarning(cls, message: str) -> None:
+        """Print the warning, without context"""
+        print(YELLOW + f"INFO: {message}" + RESET_TO_DEFAULT)
```
Comment (jnussbaum) [inline src/dsp_tools/models/custom_warnings.py:44]: Here, the mix between + concatenation and f-string is especially unelegant.

**Review by jnussbaum — APPROVED** [review]

File: src/dsp_tools/models/custom_warnings.py (hunk 6 above)
Comment (Nora-Olivia-Ammann) [reply src/dsp_tools/models/custom_warnings.py:44]: what I like about it is that it clearly and visibly separates what the user message is and what is configuration

**Pull request merged**